project_euler/problem_148/sol1.py

```diff
@@ -0,0 +1,77 @@
+"""
+Project Euler Problem 148 : https://projecteuler.net/problem=148
+Author:	Sai Teja Manchi
+Problem Statement:
+We can easily verify that none of the entries in the
+first seven rows of Pascal's triangle are divisible by 7:
+                               1
+                          1          1
+                     1          2          1
+                1          3          3          1
+           1          4          6          4          1
+      1          5         10         10          5          1
+1          6         15         20         15          6          1
+However, if we check the first one hundred rows, we will find that
+only 2361 of the 5050 entries are not divisible by 7.
+Find the number of entries which are not divisible by 7
+in the first one billion (109) rows of Pascal's triangle.
+
+Solution:
+We iteratively generate each row in the pascal triangle one-by-one.
+Since Pascal's triangle is vertically-symmetric,
+We only need to generate half of the values.
+We then count the values which are not divisible by 7.
+We only store the remainders(when divided by 7) in the list to reduce memory usage.
+
+Note: In the original problem, we need to calculate for 10^9 rows
+      but we took 10^4 rows here by default.
+"""
+
+
+def solution(pascal_row_count: int = 10**9) -> int:
+    """
```

Comment:
use below codes for better efficiency which will solve the timeout issue:-

    

      def solution(pascal_row_count: int = 10**9) -> int:
          """
          To evaluate the solution, use solution()
          >>> solution(3)
          6
          >>> solution(10)
          40
          >>> solution(100)
          2361
          """
      
          # Helper function to compute the number of odd binomial coefficients in a row
          def odd_binomials(n: int) -> int:
              res = 0
              while n > 0:
                  res += n % 2
                  n //= 2
              return res
      
          # Compute the sum of odd binomial coefficients modulo 7 for each row up to pascal_row_count
          result = 0
          for k in range(pascal_row_count):
              result += odd_binomials(k) % 7
      
          return result

Reply:
also add docs in `odd_binomials`, you can put above comment of that function inside as a doc.